src/index.ts

```diff
@@ -1,220 +1,142 @@
+import { did } from "./wallet/common.js"
 
-import { USE_WALLET_AUTH_IMPLEMENTATION } from "./auth/implementation"
-import { hasProp } from "./common"
 
 
+// 🛳
 
-// ⛰
 
+export const components = {
+  crypto(settings: Configuration & {
+    wallet: Wallet.Implementation
+  }): Promise<Crypto.Implementation> {
+    return BaseCrypto.implementation(settings.wallet, {
+      storeName: namespace(settings),
+      exchangeKeyName: "exchange-key",
+      writeKeyName: "write-key"
+    })
+  },
 
-export const ROOT_PERMISSIONS = { fs: { private: [ path.root() ], public: [ path.root() ] } }
-export const READ_KEY_PATH = path.file(path.Branch.Public, ".well-known", "read-key")
-
-
-
-// 🌳
-
-
-export type AppOptions = {
-  onAccountChange?: (appState: AppState) => unknown
-  onDisconnect?: (...args: unknown[]) => unknown
-  resetWnfs?: boolean
-  useWnfs?: boolean
+  manners(settings: Configuration & {
+    crypto: Crypto.Implementation
+    wallet: Wallet.Implementation
+  }): Promise<Manners.Implementation> {
+    return BaseManners.implementation(
+      settings.wallet,
+      { configuration: settings }
+    )
+  }
 }
 
 
 
 // 🚀
 
 
-export async function app(options?: AppOptions): Promise<AppState> {
-  let fs
+export type Options = {
+  onAccountChange?: (program: Program) => unknown
+  onDisconnect?: () => unknown
 
-  options = options || {}
-
-  const { resetWnfs = false, useWnfs = true } = options
+  wallet?: Wallet.Implementation
+}
 
-  setImplementations(USE_WALLET_AUTH_IMPLEMENTATION)
 
-  // Check if browser is supported
-  if (hasProp(self, "isSecureContext") && self.isSecureContext === false) throw InitialisationError.InsecureContext
-  if (await isSupported() === false) throw InitialisationError.UnsupportedBrowser
+/**
+ * 🚀 Build a webnative-walletauth program.
+ *
+ * Contrary to a regular webnative program,
+ * this'll create an account for you automatically.
+ * It also does session management for you.
+ *
+ * You may pass in a custom `wallet` implementation
+ * if you'd like to use another wallet than the built-in
+ * Ethereum one.
+ *
+ * See [webnative's `program`](https://webnative.fission.app/functions/program-1.html) function documentation for more info.
+ */
+export async function program(settings: Options & Partial<Components> & Configuration): Promise<Program> {
+  const wallet = settings.wallet || EthereumWallet.implementation
+  const walletCrypto = await components.crypto({ ...settings, wallet })
+
+  const defaultCrypto = await Webnative.defaultCryptoComponent(settings)
+  const manners = await components.manners({ ...settings, wallet, crypto: defaultCrypto })
+
+  // Create Webnative Program
+  const webnativeProgram = await Webnative.program({
+    ...settings,
+    crypto: defaultCrypto,
+    manners,
+  })
 
-  // Authenticate & create user if necessary
+  // Destroy existing session if wallet account changed
+  const authStrategy = webnativeProgram.auth
   const username = await wallet.username()
```

Comment:
I know it isn't technically part of this PR, but I wonder if this should be called `wallet.address()` 🤔 since it is using the wallet's address as the `username`, right? Not a blocker, but maybe something to discuss 🤷🏼

Reply:
Made an issue for that: https://github.com/fission-codes/webnative-walletauth/issues/13

Reply:
Thanks!